### ISSUE TITLE
Required Go version in README is wrong

### ISSUE THREAD

**Issue by aa2811** [prometheus/prometheus#5455]
## Bug Report
**What did you do?**

Tried to `go_get` promtool in the golang:1.11.5 container. According to the [README](https://github.com/prometheus/prometheus/blob/4d7417f5d4681cc09b7772be0d01bca693227913/README.md#building-from-source), go 1.11+ should be sufficient to do this.

**What did you expect to see?**
Sucessful docker build

**What did you see instead? Under which circumstances?**
```
Step 2/14 : RUN GOOS=linux GOARCH=amd64 CGO_ENABLED=0 go get -u github.com/prometheus/prometheus/cmd/promtool
 ---> Running in ebf3f42eb56d
# github.com/prometheus/prometheus/vendor/github.com/prometheus/tsdb/goversion
src/github.com/prometheus/prometheus/vendor/github.com/prometheus/tsdb/goversion/init.go:17:9: undefined: _SoftwareRequiresGOVERSION1_12
The command '/bin/sh -c GOOS=linux GOARCH=amd64 CGO_ENABLED=0 go get -u github.com/prometheus/prometheus/cmd/promtool' returned a non-zero code: 2
```
**Environment**

golang:1.11.5 container. 

This step worked in golang:1.12.3, and judging by [this file](https://github.com/prometheus/prometheus/blob/4cce6291ae006a5b85e73f308cc0b55e313c150d/vendor/github.com/prometheus/tsdb/goversion/goversion.go) which has recently been added, I think this is because you now need go 1.12+ to build from source, so the README needs updating.

**Comment by simonpasquier** [comment]
Do you want to submit the PR for this?

**Comment by cstyan** [comment]
fixed in #5471
